packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -417,13 +448,24 @@ - (void)setupEventSinkIfReadyToPlay {
     AVAsset *asset = currentItem.asset;
     if ([asset statusOfValueForKey:@"tracks" error:nil] != AVKeyValueStatusLoaded) {
       void (^trackCompletionHandler)(void) = ^{
-        if ([asset statusOfValueForKey:@"tracks" error:nil] != AVKeyValueStatusLoaded) {
+        NSError *error;
+        switch ([asset statusOfValueForKey:@"tracks" error:&error]) {
+          case AVKeyValueStatusLoaded:
+            // This completion block will run on an AVFoundation background queue.
+            // Hop back to the main thread to set up event sink.
+            [self performSelector:_cmd
+                         onThread:NSThread.mainThread
+                       withObject:self
+                    waitUntilDone:NO];
+            break;
           // Cancelled, or something failed.
-          return;
+          case AVKeyValueStatusFailed:
+            // if something failed then future should result in error
+            [self sendFailedToLoadVideoEvent:error];
```

Comment:
can you also specify which thread this is on

Reply:
On some AVFoundation background queue (this is also mentioned in a comment a few lines above):

https://developer.apple.com/documentation/avfoundation/avasynchronouskeyvalueloading/1387321-loadvaluesasynchronouslyforkeys